hw6/counter.py

```diff
@@ -0,0 +1,40 @@
+"""
+Написать декоратор instances_counter, который применяется к любому классу
+и добавляет ему 2 метода:
+get_created_instances - возвращает количество созданых экземпляров класса
+reset_instances_counter - сбросить счетчик экземпляров,
+возвращает значение до сброса
+Имя декоратора и методов не менять
+Ниже пример использования
+"""
+
+
+def instances_counter(cls):
+    """
+    Decorator for counting instances of a class
+    Creates new __new__ method and assign it to existing class
+    Adds get_created_instances and reset_instances_counter methods to existing class
+    """
+    cls.created_instances = 0
+
+    def __new__(cls):
+        obj = super(cls, cls).__new__(cls)
+        cls.created_instances += 1
```

Comment:
What if:
```
@isntances_counter()
class A:
    pass


a = A()
a. created_instances = 123
```
and how to prevent this?

Reply:
There is no way to prevent this. We only can point to programmer with naming _created_instances that we do not want him to touch this attribute. So I renamed it with using "_" convention - Done.